### PR TITLE
Whitelisting of enclaves 

### PR DESCRIPTION
**Pull request by felixfaisal**
This PR does the following: 
- Create new extrinsic for whitelisting enclaves 
- `Storage Map` for storing whitelisted enclaves 
- `Errors` and `Events` for the above 
- `Ensure!` in register_enclave to ensure enclave_account_id is whitelisted 

Things yet to do: 
- [x] Unit Testing

**Please Note**: I would like to get the logic approved before going about writing the unit tests. 

**Update**: Logic verified by @Gauthamastro, Waiting for other comments

### CODE REVIEW

**Review by 35359595 — DISMISSED** [review]
LGTM

File: pallets/ocex/src/lib.rs [hunk 1]
```diff
@@ -649,6 +658,20 @@ pub mod pallet {
 			debug!("registered enclave at time =>{:?}", report.timestamp);
 			Ok(())
 		}
+
+		/// In order to register itself - enclave accountid must be whitelisted called by
+		/// Governance
```
Comment (serhii-temchenko) [inline pallets/ocex/src/lib.rs]: Please update to:
```rust
		/// In order to register itself - enclave account id must be whitelisted and called by
		/// Governance
```

File: pallets/ocex/src/lib.rs [hunk 2]
```diff
@@ -649,6 +658,20 @@ pub mod pallet {
...
+		#[pallet::weight(<T as Config>::WeightInfo::register_enclave())]
+		pub fn whitelist_enclave(
+			origin: OriginFor<T>,
+			enclave_account_id: T::AccountId,
+		) -> DispatchResult {
+			T::GovernanceOrigin::ensure_origin(origin)?;
+			// It will just overwrite if account_id is already whitelisted
+			<WhitelistedEnclaves<T>>::insert(&enclave_account_id, true);
+			Self::deposit_event(Event::EnclaveWhitelisted(enclave_account_id.clone()));
```
Comment (serhii-temchenko) [inline pallets/ocex/src/lib.rs]: I assume `clone` can be removed here.

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

File: pallets/ocex/src/lib.rs (hunk 1 above)
Comment (felixfaisal) [reply pallets/ocex/src/lib.rs]: Done ✅

File: pallets/ocex/src/lib.rs (hunk 2 above)
Comment (felixfaisal) [reply pallets/ocex/src/lib.rs]: Done ✅

File: pallets/ocex/src/lib.rs (hunk 1 above)
Comment (serhii-temchenko) [reply pallets/ocex/src/lib.rs]: @felixfaisal I don't see any changes about this one...

Comment (felixfaisal) [reply pallets/ocex/src/lib.rs]: This has been updated now, Sorry for the confusion

**Review by serhii-temchenko — APPROVED** [review]

**Review by Gauthamastro — APPROVED** [review]

**Pull request merged**